Updates Doc for Source*Id (#2284)
FusionAuth/fusionauth-site: site/docs/v1/tech/apis/themes.adoc

This API is used to create a new Theme.

> * link:#create_theme_with_id[Create a Theme with a randomly generated Id]

Reviewer: ```suggestion
* link:#create_theme_generate_id[Create a Theme with a randomly generated Id]
```

include::docs/v1/tech/apis/_theme-response-body-suffix.adoc[]
{%capture singleResponse %}false{% endcapture %}

== Create a New Theme From an Existing Theme

This API is used to create a new Theme from an existing Theme.

=== Request

[.api-authentication]

> link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Create a new Theme with a randomly generated Id

Reviewer: ```suggestion
link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Create a new Theme from an existing Theme with a generated Id
```

Author: We use `generated` and `randomly generated` interchangeably.  Is there a preference here?

[.endpoint]
.URI
--
[method]#POST# [uri]#/api/theme#
--

[.api-authentication]

> link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Create a Theme with the provided unique Id

Reviewer: ```suggestion
link:/docs/v1/tech/apis/authentication#api-key-authentication[icon:lock[type=fas]] Create a new Theme from an existing Theme with the given Id
```

[.endpoint]
.URI
--
[method]#POST# [uri]#/api/theme/`\{themeId\}`#
--

==== Request Parameters

[.api]
[field]#themeId# [type]#[UUID]# [optional]#Optional# [default]#defaults to secure random UUID#::
The Id to use for the new Theme. If not specified a secure random UUID will be generated.

==== Request Body

[.api]
[field]#sourceThemeId# [type]#[UUID]# [required]#Required#::

> The Id of an existing Theme to make a copy of.

Reviewer: ```suggestion
The Id of an existing Theme from which a copy will be made.
```

+

> If present, the [field]#defaultMessages#, [field]#localizedMessages#, [field]#templates#, and [field]#stylesheet# from the source Theme will be copied to the new Theme. All other fields not listed here may be provided on this request, such as [field]#theme.name#.  The [field]#theme.name# is required.

Reviewer: I don't think `theme.name` should be mentioned here because it's got another field listed on the request body.

The "if present" doesn't make sense anymore because we are calling this out as a separate API, so `sourceThemeId` will always be present when using this API.

Are there other fields that can be specified? It seems like the only other field is `data`, which isn't worth calling out IMO.

```suggestion
The [field]#defaultMessages#, [field]#localizedMessages#, [field]#templates#, and [field]#stylesheet# from the source Theme will be copied to the new Theme.
```

Author: I agree - `data` can be left out.

[field]#theme.name# [type]#[String]# [required]#Required#::
A unique name for the Theme.

.Example request JSON
[source,json]

> {
> 	"sourceThemeId": "64773453-bb11-457b-a3d6-7475ec2259d0",
> 	"theme": {
> 		"name": "Orange Theme - copied"
> 	}
> }

Reviewer: ```suggestion
{
  "sourceThemeId": "64773453-bb11-457b-a3d6-7475ec2259d0",
  "theme": {
    "name": "Orange Theme - copied"
  }
}
```
